In [1]:
import pandas as pd
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.by import By
import time, urllib.request
import requests


In [11]:
username = []
times = []
comments = []
likes = []
LINK_INSTAGRAM = "https://www.instagram.com"
LINK_POST = "https://www.instagram.com/p/Cu4iHlRsBed/"

In [3]:
#C:\Program Files\Google\Chrome\Application
#r"C:\download\chromedriver_win32\chromedriver.exe"
#PATH = r"D:\Program Files\chrome\chromedriver.exe"
driver = webdriver.Chrome()

In [4]:
driver.get("https://www.instagram.com/")

# Login session (automate the login and save login info)

In [5]:
#login
time.sleep(5)
username=driver.find_element(By.NAME, 'username')
password=driver.find_element(By.NAME, 'password')
#username=driver.find_element_by_css_selector("input[name='username']")
#password=driver.find_element_by_css_selector("input[name='password']")



In [6]:
time.sleep(5)
username.clear()
password.clear()


In [7]:
time.sleep(5)
username.send_keys("username")
password.send_keys("password")
#login = driver.find_element_by_css_selector("button[type='submit']").click()
login = driver.find_element(By.XPATH, "//form[@id='loginForm']/div[1]/div[3]/button[1]").click()

In [8]:
#save your login info?
time.sleep(5)
notnow = driver.find_element(By.XPATH, "//div[@role='button']").click()


In [9]:
time.sleep(5)
notnow2 = driver.find_element(By.XPATH,"//button[contains(text(), 'Not Now')]").click()

In [12]:
time.sleep(5)
driver.get(LINK_POST)

### SCRAPING DATA POST

In [13]:
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys

## Comment section class
"x9f619.x78zum5.xdt5ytf.x5yr21d.xexx8yu.x1pi30zi.x1l90r2v.x1swvt13.x10l6tqk.xh8yej3"
x9f619 x78zum5 xdt5ytf x5yr21d xexx8yu x1pi30zi x1l90r2v x1swvt13 x10l6tqk xh8yej3

"x5yr21d.xw2csxc.x1odjw0f.x1n2onr6"

## all comment not scroll class 
"x78zum5 xdt5ytf x1iyjqo2"

In [14]:
# Comment section scrollable
time.sleep(5)
comment_section = driver.find_element(By.CLASS_NAME, "x5yr21d.xw2csxc.x1odjw0f.x1n2onr6")
print(comment_section)

<selenium.webdriver.remote.webelement.WebElement (session="2f7952c181ce772533f7d32039a9f406", element="F74EBF5D2A8C0EB8CD1F9F5223E18D38_element_70")>


In [15]:
time.sleep(10)

verical_ordinate = 100
for i in range(0, 60):
    print(verical_ordinate)
    driver.execute_script("arguments[0].scrollTop = arguments[1]", comment_section, verical_ordinate)
    verical_ordinate += 400
    time.sleep(2)
#elem = driver.find_element(By.TAG_NAME, "html")
#time.sleep(5)
#comment_section.send_keys(Keys.END)
#actions = ActionChains(driver)
#actions.move_to_element(element).perform()

100
300
500
700
900
1100
1300
1500
1700
1900
2100
2300
2500
2700
2900
3100
3300
3500
3700
3900
4100
4300
4500
4700
4900
5100
5300
5500
5700
5900
6100
6300
6500
6700
6900
7100
7300
7500
7700
7900
8100
8300
8500
8700
8900
9100
9300
9500
9700
9900
10100
10300
10500
10700
10900
11100
11300
11500
11700
11900


In [16]:
# all comment <div> elements
time.sleep(5)
all_comment = driver.find_elements(By.CLASS_NAME, "x78zum5.xdt5ytf.x1iyjqo2")
print(len(all_comment))

110


In [17]:
for i in range(0, len(all_comment)):
    print(all_comment[i].text)


playapex
•
Follow
playapex
 1w
An update on Ranked: What we’ve learned so far with Arsenal and a peek at what’s to come.

Click the link in our bio to read more.
deshawn.hw
 1w
Use me as a “everyone gonna complain no matter what” button
2,200 likes
Reply
View all 58 replies
prizzlethegreat
 1w
Use me as a "Doubt" button
764 likes
Reply
View all 3 replies
no.mans.guy
 1w
Please make kills and squad wipes mean something
446 likes
Reply
View all 14 replies
herodelta
 1w
I like how they greyed out the master emblem, they know 90% of the people who have it now don’t deserve it
678 likes
Reply
View all 41 replies
itsbriangoon
 1w
This didn’t tell us anything besides the fact that a lot of you need to be fired
74 likes
Reply
View all 10 replies
husseinaljanabi
 1w
We had some trouble for 16 seasons when it comes to ranked and they were manageable. But how do you mess up rank so bad in season 17?
25 likes
Reply
View all 2 replies
brug_im_sleep
 1w
Reminder that this seasons masters means nothi

playapex
•
Follow
playapex
 1w
An update on Ranked: What we’ve learned so far with Arsenal and a peek at what’s to come.

Click the link in our bio to read more.
deshawn.hw
 1w
Use me as a “everyone gonna complain no matter what” button
2,200 likes
Reply
View all 58 replies
prizzlethegreat
 1w
Use me as a "Doubt" button
764 likes
Reply
View all 3 replies
no.mans.guy
 1w
Please make kills and squad wipes mean something
446 likes
Reply
View all 14 replies
herodelta
 1w
I like how they greyed out the master emblem, they know 90% of the people who have it now don’t deserve it
678 likes
Reply
View all 41 replies
itsbriangoon
 1w
This didn’t tell us anything besides the fact that a lot of you need to be fired
74 likes
Reply
View all 10 replies
husseinaljanabi
 1w
We had some trouble for 16 seasons when it comes to ranked and they were manageable. But how do you mess up rank so bad in season 17?
25 likes
Reply
View all 2 replies
brug_im_sleep
 1w
Reminder that this seasons masters means nothi

playapex
 1w
An update on Ranked: What we’ve learned so far with Arsenal and a peek at what’s to come.

Click the link in our bio to read more.
deshawn.hw
 1w
Use me as a “everyone gonna complain no matter what” button
2,200 likes
Reply
View all 58 replies
prizzlethegreat
 1w
Use me as a "Doubt" button
764 likes
Reply
View all 3 replies
no.mans.guy
 1w
Please make kills and squad wipes mean something
446 likes
Reply
View all 14 replies
herodelta
 1w
I like how they greyed out the master emblem, they know 90% of the people who have it now don’t deserve it
678 likes
Reply
View all 41 replies
itsbriangoon
 1w
This didn’t tell us anything besides the fact that a lot of you need to be fired
74 likes
Reply
View all 10 replies
husseinaljanabi
 1w
We had some trouble for 16 seasons when it comes to ranked and they were manageable. But how do you mess up rank so bad in season 17?
25 likes
Reply
View all 2 replies
brug_im_sleep
 1w
Reminder that this seasons masters means nothing and to not flex

no.mans.guy
 1w
Please make kills and squad wipes mean something
446 likes
Reply
herodelta
 1w
I like how they greyed out the master emblem, they know 90% of the people who have it now don’t deserve it
678 likes
Reply
itsbriangoon
 1w
This didn’t tell us anything besides the fact that a lot of you need to be fired
74 likes
Reply
husseinaljanabi
 1w
We had some trouble for 16 seasons when it comes to ranked and they were manageable. But how do you mess up rank so bad in season 17?
25 likes
Reply
brug_im_sleep
 1w
Reminder that this seasons masters means nothing and to not flex it like ur good or something 💀💀💀💀
146 likes
Reply
officialapexworld
 1w
Apologies.. but this explains nothing and is just a “we heard you, we get it”
624 likes
Reply
1xtitan
 1w
Cool. Now take out Storm point 🤝
87 likes
Reply
raaed.khan214
 1w
Approximately 1 million people became masters this season (17) and you are saying that you are going to “be more competitive and tighten the gap for more reasonable ranked” 

bkent016
 1w
What a blanketed statement with no actual information 🤣
2 likes
Reply
i_am_yourdevil__
 1w
This is like the 6th time yall change the rank system lmfaooo
1 like
Reply
d.endmusic
 1w
The false Masters gonna get cooked next season
1 like
Reply
_miick
 1w
NOBODY IS TALKING ABOUT THESE FUSE EXPLOITS!?!?1!? Hello?
1 like
Reply
terrysadler91
 1w
Madness this season
1 like
Reply
its.nefarious
 1w
I re-download your game, what did you do with arenas bruh
2 likes
Reply
dumpsterpapi
 1w
Fix the Fuze Ult. vehicle exploit plz
2 likes
Reply
drakenfps__
 1w
Need arenas😔
4 likes
Reply
aloy_abitch
 1w
@qwety.01 finally...
1 like
Reply
ok_dork323
 1w
Upgrade your engine, it’ll be insane
2 likes
Reply
shinexdd_
 1w
TLDR: it will suck next season as well
2 likes
Reply
brycedavis95
 1w
@__seanthesheep @franklinnwarrenn
2 likes
Reply
stefangonev
 1w
cross progression
1 like
Reply
diarios_
 1w
So still no cross progression?
1 like
Reply
manuel__andrade
 1w
They said a whole lot of nothing
Reply


## Small Explanation
For scraping data with selenium you either need to use attributes from tag componets such as id, class, name, and etc.
From experimenting with this notebooks, I find filtering data with class is the easiest way since the attribute class
in components is unique, I collected class attribute in below explanation.

### Username class

x9f619.xjbqb8w.x1rg5ohu.x168nmei.x13lgxp2.x5pf9jr.xo71vjh.x1n2onr6.x1plvlek.xryxfnj.x1c4vz4f.x2lah0s.x1q0g3np.xqjyukv.x6s0dn4.x1oa3qoh.x1nhvcw1


### Time class
"x76ihet.xwmqs3e.x112ta8.xxxdfa6.x1roi4f4.xexx8yu.x4uap5.x18d9i69.xkhd6sd.x1n2onr6"

### Comment class
"x1lliihq.x1plvlek.xryxfnj.x1n2onr6.x193iq5w.xeuugli.x1fj9vlw.x13faqbe.x1vvkbs.x1s928wv.xhkezso.x1gmr53x.x1cpjm7i.x1fgarty.x1943h6x.x1i0vuye.xvs91rp.xo1l8bm.x5n08af.x10wh9bi.x1wdrske.x8viiok.x18hxmgj"

x1lliihq x1plvlek xryxfnj x1n2onr6 x193iq5w xeuugli x1fj9vlw x13faqbe x1vvkbs x1s928wv xhkezso x1gmr53x x1cpjm7i x1fgarty x1943h6x x1i0vuye xvs91rp xo1l8bm x5n08af x10wh9bi x1wdrske x8viiok x18hxmgj

### love/like class
"x1lliihq.x1plvlek.xryxfnj.x1n2onr6.x193iq5w.xeuugli.x1fj9vlw.x13faqbe.x1vvkbs.x1s928wv.xhkezso.x1gmr53x.x1cpjm7i.x1fgarty.x1943h6x.x1i0vuye.x1fhwpqd.x1s688f.x1roi4f4.x1s3etm8.x676frb.x10wh9bi.x1wdrske.x8viiok.x18hxmgj"

In [18]:
#Load username
time.sleep(5)
load_username = driver.find_elements(By.CLASS_NAME, "x9f619.xjbqb8w.x1rg5ohu.x168nmei.x13lgxp2.x5pf9jr.xo71vjh.x1n2onr6.x1plvlek.xryxfnj.x1c4vz4f.x2lah0s.x1q0g3np.xqjyukv.x6s0dn4.x1oa3qoh.x1nhvcw1")
print(len(load_username))

106


In [19]:
#Load time
time.sleep(5)
load_time = driver.find_elements(By.CLASS_NAME, "x76ihet.xwmqs3e.x112ta8.xxxdfa6.x1roi4f4.xexx8yu.x4uap5.x18d9i69.xkhd6sd.x1n2onr6")
print(len(load_time))

105


In [20]:
#Load comment
time.sleep(5)
load_comment = driver.find_elements(By.CLASS_NAME, "x1lliihq.x1plvlek.xryxfnj.x1n2onr6.x193iq5w.xeuugli.x1fj9vlw.x13faqbe.x1vvkbs.x1s928wv.xhkezso.x1gmr53x.x1cpjm7i.x1fgarty.x1943h6x.x1i0vuye.xvs91rp.xo1l8bm.x5n08af.x10wh9bi.x1wdrske.x8viiok.x18hxmgj")
print(len(load_comment))

213


In [22]:
#Load comment
time.sleep(5)
load_comment = driver.find_elements(By.XPATH, "//div[@class='x9f619 xjbqb8w x78zum5 x168nmei x13lgxp2 x5pf9jr xo71vjh x1uhb9sk x1plvlek xryxfnj x1c4vz4f x2lah0s xdt5ytf xqjyukv x1cy8zhl x1oa3qoh x1nhvcw1']")
print(len(load_comment))

105


In [23]:
#Load like
time.sleep(5)
load_like = driver.find_elements(By.XPATH, "//div[@class='x1i10hfl xjbqb8w x6umtig x1b1mbwd xaqea5y xav7gou x9f619 x1ypdohk xt0psk2 xe8uvvx xdj266r xat24cr x1mh8g0r xexx8yu x4uap5 x18d9i69 xkhd6sd x16tdsg8 x1hl2dhg xggy1nq x1a2a7pz xq8finb x87ps6o x1d5wrs8']")
print(len(load_like))

206


as you can see there is different length of data we gathered, let's explore the data with code below.

Total username we gathered is as an example 106, but if we look closely to the data the first data is actually original poster, so we skip the looping by 1.

In [25]:
username = []
for i in range(1, len(load_username)):
    print(load_username[i].text)
    username.append(load_username[i].text)

deshawn.hw
prizzlethegreat
no.mans.guy
herodelta
itsbriangoon
husseinaljanabi
brug_im_sleep
officialapexworld
1xtitan
raaed.khan214
apex.wiz
mayor_of_apex
_marshall_mxl
basicallymarii
mytechnicalities
hhx_3dart
santihawk
jrfromthetrapspam
king.quani392
planetdab
wickgeek
gsviesa
oisin.j.king
planetdab
owo_bolt
_.robotickz
dannymondelli
soarkobi
jerome761993
bertoqr.ig
joshinya17
justin.sann
4kworldwide
v0.iid
thisdudeiswaysilly
mkgwayflight
rawrrrbyn
r00sterc0gburne
knife15_
laroyce._
thicknmessy
sambangparinda
matijjas
sami_berta04
oneofmany33
stolencrucifix87
jalrxn
just.for.save.clip
hydracat88
adrian_g_mcloving
___christiancollins
pastymexican714
armand__es
gl1tch.chersi
owen.p.10
jakehall27
whittythewanderer
jaydub4587
plmpmyomelette
mohammed.chucha
_mvnu_
apex_memextuve
dsczaplicki
herbi_111
ptrjsphrvra
sheluvmarc
bkent016
i_am_yourdevil__
d.endmusic
_miick
terrysadler91
its.nefarious
dumpsterpapi
drakenfps__
aloy_abitch
ok_dork323
shinexdd_
brycedavis95
stefangonev
diarios_
manu

In [26]:
#.get_attribute("datetime")
for i in range(0, len(load_comment)):
    print(load_comment[i].text)
    comments.append(load_comment[i].text)

Use me as a “everyone gonna complain no matter what” button
Use me as a "Doubt" button
Please make kills and squad wipes mean something
I like how they greyed out the master emblem, they know 90% of the people who have it now don’t deserve it
This didn’t tell us anything besides the fact that a lot of you need to be fired
We had some trouble for 16 seasons when it comes to ranked and they were manageable. But how do you mess up rank so bad in season 17?
Reminder that this seasons masters means nothing and to not flex it like ur good or something 💀💀💀💀
Apologies.. but this explains nothing and is just a “we heard you, we get it”
Cool. Now take out Storm point 🤝
Approximately 1 million people became masters this season (17) and you are saying that you are going to “be more competitive and tighten the gap for more reasonable ranked” lmao that’s a joke Fr fr 🤣🤣🤣. Just bring Arenas back and we will call it settled.
In the dev blog you say “These solutions together will slow the pace to Maste

In [68]:
#.get_attribute("datetime")
times =[]
for i in range(0, len(load_time)):
    print(load_time[i].get_attribute("datetime"))
    times.append(load_time[i].get_attribute("datetime"))

2023-07-19T15:07:18.000Z
2023-07-19T15:06:50.000Z
2023-07-19T15:14:59.000Z
2023-07-19T15:14:18.000Z
2023-07-19T15:06:04.000Z
2023-07-19T15:09:04.000Z
2023-07-19T15:11:18.000Z
2023-07-19T15:13:32.000Z
2023-07-19T16:30:27.000Z
2023-07-19T15:15:44.000Z
2023-07-19T15:21:10.000Z
2023-07-19T15:22:17.000Z
2023-07-19T15:15:35.000Z
2023-07-19T15:06:51.000Z
2023-07-19T15:15:02.000Z
2023-07-19T16:32:25.000Z
2023-07-22T03:04:53.000Z
2023-07-19T15:12:47.000Z
2023-07-19T15:07:40.000Z
2023-07-19T16:28:43.000Z
2023-07-19T15:07:07.000Z
2023-07-25T03:05:45.000Z
2023-07-19T15:17:44.000Z
2023-07-19T15:11:59.000Z
2023-07-19T15:15:39.000Z
2023-07-19T15:11:57.000Z
2023-07-19T15:16:50.000Z
2023-07-21T05:34:35.000Z
2023-07-19T15:15:10.000Z
2023-07-19T15:06:59.000Z
2023-07-19T15:13:18.000Z
2023-07-19T20:43:11.000Z
2023-07-19T15:58:56.000Z
2023-07-19T15:18:41.000Z
2023-07-20T09:48:56.000Z
2023-07-19T15:14:37.000Z
2023-07-19T15:20:21.000Z
2023-07-19T15:17:36.000Z
2023-07-21T05:33:24.000Z
2023-07-19T15:25:13.000Z


In [28]:
for i in range(0, len(load_like)):
    print(load_like[i].text)

2,200 likes
Reply
764 likes
Reply
446 likes
Reply
678 likes
Reply
74 likes
Reply
25 likes
Reply
146 likes
Reply
624 likes
Reply
87 likes
Reply
63 likes
Reply
26 likes
Reply
434 likes
Reply
47 likes
Reply
96 likes
Reply
37 likes
Reply
10 likes
Reply
3 likes
Reply
28 likes
Reply
82 likes
Reply
1 like
Reply
53 likes
Reply
2 likes
Reply
1 like
Reply
33 likes
Reply
1 like
Reply
12 likes
Reply
1 like
Reply
1 like
Reply
17 likes
Reply
4 likes
Reply
13 likes
Reply
3 likes
Reply
2 likes
Reply
1 like
Reply
Reply
4 likes
Reply
9 likes
Reply
1 like
Reply
2 likes
Reply
17 likes
Reply
3 likes
Reply
Reply
16 likes
Reply
8 likes
Reply
1 like
Reply
12 likes
Reply
1 like
Reply
1 like
Reply
1 like
Reply
1 like
Reply
2 likes
Reply
2 likes
Reply
2 likes
Reply
5 likes
Reply
4 likes
Reply
1 like
Reply
1 like
Reply
Reply
2 likes
Reply
3 likes
Reply
4 likes
Reply
1 like
Reply
14 likes
Reply
2 likes
Reply
Reply
1 like
Reply
2 likes
Reply
1 like
Reply
1 like
Reply
1 like
Reply
1 like
Reply
2 likes
Reply
2 likes


There is some notes whene scrapping likes data from instagram, if the user have 0 likes instead showing 0 likes, it shows only "Reply", AND  it shows "see translation" in that components area. check code below to fill the missing data.

In [62]:
likes=[]
for i in range(0, len(load_like)):
    if(load_like[i].text == "Reply"):
        if(i < len(load_like)-1 and load_like[i+1].text == "Reply"):
            likes.append(0)
        continue
    if(load_like[i].text == "See translation"):
        continue
    else:
        txt = load_like[i].text
        x = txt.split()
        likes.append(x[0])

In [69]:
print(len(username))
print(len(comments))
print(len(likes))
print(len(times))

105
105
105
105


In [70]:
data = {"username":username, "comments": comments, "likes":likes, "datetime": times}

In [71]:
df = pd.DataFrame(data)
df

username                                           comments  \
0          deshawn.hw  Use me as a “everyone gonna complain no matter...   
1     prizzlethegreat                         Use me as a "Doubt" button   
2         no.mans.guy   Please make kills and squad wipes mean something   
3           herodelta  I like how they greyed out the master emblem, ...   
4        itsbriangoon  This didn’t tell us anything besides the fact ...   
..                ...                                                ...   
100     chriszevgolis                Hahaha what a joke. Game is dead af   
101  suvansh_pathania                        bring 120FPS ON CONSOLEEEE😢   
102         rell.1990                                            W 🏌🏾‍♂️   
103           zayney_                                        fix control   
104   l3rianl3ezuijen                            We will see about that.   

     likes                  datetime  
0    2,200  2023-07-19T15:07:18.000Z  
1      764  2023-07-19T15:06:50.000Z  
2      446  2023-07-19T15:14:59.000Z  
3      678  2023-07-19T15:14:18.000Z  
4       74  2023-07-19T15:06:04.000Z  
..     ...                       ...  
100      1  2023-07-19T15:13:32.000Z  
101      1  2023-07-19T15:13:43.000Z  
102      1  2023-07-19T15:13:39.000Z  
103      1  2023-07-19T15:13:21.000Z  
104      1  2023-07-19T15:10:23.000Z  

[105 rows x 4 columns]

In [85]:

df.to_csv('Instagram-post-Cu4iHlRsBed.csv', index=False)

#df.to_excel('Instagram-post-'+LINK_POST+'.excel', index=False)

In [ ]:
driver.close()
driver.quit()